extensions/objects/templates/math-editor.component.spec.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the math editor component.
+ */
+
+describe('MathEditor', function() {
+  var MathEditorCtrl = null;
+  class MockGuppy {
+    constructor(id: string, config: Object) {}
+
+    event(name: string, handler: Function): void {
+      handler();
+    }
+    asciimath(): string {
+      return 'Dummy value';
+    }
+    render(): void {}
+  }
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.inject(function($componentController) {
+    MathEditorCtrl = $componentController('mathEditor');
+    (<any>window).Guppy = MockGuppy;
```

Comment:
This doesn't seem right. Can you get @nishantwrp @vojtechjelinek signoff?

Also, shouldn't we be using $window instead?

Reply:
@iamprayush Why is any needed here? We have typings for window in `typings` directory.

Reply:
> Also, shouldn't we be using $window instead?

Yeah that works as well and we won't need to use `any` with that. I've changed it now, thanks!